Swinging object

AdSand opened the pull request:
Added a swinging object obstacle to the game. At the minute, doesn't damage the player or anything. Works across server.

Simply add the swinging object component and Voilà! The object will swing.

Use SetRadius(float) to change how much the swing is. Currently moves at one speed, but we can easily change this later on

AdSand committed:
Added component onto a block

AdSand committed:
Changed order to now update correct object

AdSand committed:
Object moving in circle

AdSand committed:
Fixed issue, move better

AdSand committed:
Swinging object working. Waiting for start update

AdSand committed:
Added setter for radius

Odycious commented on Replicated/Components/SwingingObject.cpp:
```diff
@@ -0,0 +1,27 @@
+    gameObject->GetTransform().SetPosition(tempPos);
+
+
+    std::cout << tempPos.y << std::endl;
```
I have removed this cout statement

Odycious commented on Replicated/Replicated.cpp:
```diff
@@ -26,6 +26,17 @@ void Replicated::AddBlockToLevel(GameObject *g, GameWorld& world, PrimitiveGameO
+void Replicated::AddSwingingBlock(GameObject* g, GameWorld& world) {
+
+    world.AddGameObject(g, true);
+    auto volume = new AABBVolume(Vector3(10, 10, 10) * 0.5f);
```
This should probably be changed from flat values, different swinging objects may require different sizes maybe?

Odycious commented on Server/States/RunningState.cpp:
```diff
@@ -198,5 +200,13 @@ void RunningState::BuildLevel(const std::string &levelName)
-    AddTriggersToLevel();
+
+    // TEST SWINGING OBJECT ON THE SERVER
+    auto swingingObject = new GameObject("YEEEEHAWWWWW");
```
I've changed the name to Swinging_Object

Odycious committed:
Minor changes

Removed cout statement, changed the swinging object name, minor spacing changes

Odycious committed:
Merge branch 'master' into Swinging-Object
